test/end2end_test.go

```diff
@@ -4238,6 +4243,73 @@ func testClientRequestBodyErrorCancelStreamingInput(t *testing.T, e env) {
 	})
 }
 
+func TestClientResourceExhaustedCancelFullDuplex(t *testing.T) {
+	defer leakcheck.Check(t)
+	for _, e := range listTestEnv() {
+		if e.httpHandler {
+			// httpHanlder write won't be blocked on flow control window.
+			continue
+		}
+		testClientResourceExhaustedCancelFullDuplex(t, e)
+	}
+}
+
+func testClientResourceExhaustedCancelFullDuplex(t *testing.T, e env) {
+	te := newTest(t, e)
+	recvErr := make(chan error, 1)
+	timeoutErr := errors.New("1s timeout reached")
+	ts := &funcServer{fullDuplexCall: func(stream testpb.TestService_FullDuplexCallServer) error {
+		_, err := stream.Recv()
+		if err != nil {
+			t.Fatalf("expecting <nil> error, got %v", err)
```

Comment:
`t.Fatalf` doesn't actually terminate the test if it's done in a separate goroutine.

If you return the error instead, your test will fail anyway because of the other checks.  Just make sure you do a "defer close(recvErr)" at the top of this function.